src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = sourceString
+    .split(';')
+    .map((item) => item.split(':'))
+    .map((item) => item.map((value) => value.trim()))
+    .map((item) => item.filter((value) => value !== ''))
+    .filter((item) => item.length !== 0)
+    .reduce((object, item) => {
+      object[item[0]] = item[1];
+
+      return object;
+    }, {});
+
+  return result;
```

Comment:
Avoid using loops in this task as per the checklist. The `reduce` method is acceptable here since it's being used to transform the array into an object, which is not a typical loop operation.